### DISCUSSION TITLE
How i can read the ZipInputStream in the second, third and so on times?

### DISCUSSION THREAD

**Discussion by sovetkin** [icsharpcode/SharpZipLib#696] [Q&A] [answered]
Hello, everyone!

I try to parse the excel files. I want to get all worksheets as a ZipEntry, store them separately and then read from the used stream particular entries using the following code:

```
if (zipEntry.CanDecompress)
{
       long length = zipEntry.Size;
       long offset = zipEntry.Offset;
       byte[] data = new byte[length];

       streamSize = zipInputStream.Read(data, (int)offset, data.Length);
}
```

but it doesn't work. It works if you iterate through the whole archive. I don't want to do this every time when I need to get access to a particular entity.

How I can solve this issue?

**Comment by piksel** [comment, marked as answer]
Use `ZipFile` instead of `ZipInputStream` for random access.

**Reply by sovetkin** [reply]
Ok, thanks. I'll try this way

**Reply by sovetkin** [reply]
@piksel, could you clarify Why ZipInputStream class is needed? Because I have had to refuse from this class in the end.

**Reply by piksel** [reply]
It's mainly recommended when a stream-like interface is needed, such as reading a temporary in-memory archive (or whatever use case you might have). It's a different approach, with different strengths (random access not being one of them).
